tests/components/stream/test_recorder.py

```diff
@@ -154,6 +169,37 @@ async def test_recorder_timeout(hass, hass_client, stream_worker_sync):
         await hass.async_block_till_done()
 
 
+async def test_recorder_duration_keepalive(hass, hass_client, record_worker_sync):
```

Comment:
What is this test supposed to do? The timeout is longer than source video, but keepalive is on so the stream restarts. Is this the intended behavior?

Reply:
I'd like to make the timeout happen before the end of stream, so looking for hacks to simulate that.

This approach currently hangs. I think this may be a bit hard and not worth it.

Reply:
The source video has a duration of 5 seconds and the `TEST_TIMEOUT` is 8 seconds...I guess we could either use a longer source or reduce the TEST_TIMEOUT further, but I also agree this may not be worth it.

Reply:
The other subtle point here I think I'd that keepaive will just restart and decide it again.  We don't treat "end of stream" any different from "stream had temp failure" so I'm taking advantage of that here.  The other approach is to pause the worker thread to prevent it from exiting.

One related point I've noticed while exploring this is that `stop` with it's `join` may block the hassio loop. I was considering moving it to use `asyncio` primitives instead effectively moving off of `Thread` and doing that may let me simplify some of these testing patterns, but unclear.

Anyway I'll revert this unless I get any better idea.

Reply:
Oh..I was thinking that it might be getting held up by the worker thread since it has to wait to restart. Maybe set `_fast_restart_once` so it doesn't wait?